mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerImpl.java

```diff
@@ -61,37 +61,55 @@
 @Slf4j
 class ExecutorStateManagerImpl implements ExecutorStateManager {
     private final Map<TaskExecutorID, TaskExecutorState> taskExecutorStateMap = new HashMap<>();
-    Cache<String, JobRequirements> pendingJobRequests = CacheBuilder.newBuilder()
+
+    /**
+     * Cache the available executors ready to accept assignments
+     */
+    private final Map<SchedulingConstraints, NavigableSet<TaskExecutorHolder>> executorsBySchedulingConstraints = new HashMap<>();
```

Comment:
SchedulingConstraints as key here imply the machine def needs to match the schedule request exactly which is error prone and might not be fully back compact on existing jobs not exactly matching on network/disk?

Reply:
I don't see the problem with changing network/disk and have a separate key for newly registered TEs. Assignment will be done in either the old or new set of TEs. am I missing anything?

Reply:
I think I am more referring to that SchedulingConstraints is a contract/data class used across APIs while its key (toString) here is also used as the tracking logic inside the internal of the executorManager. Thus any change on this class can impact the tracking mechanism in executorManager, which is a coupling that might cause unintentional behavior to any changes around it.